Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -0,0 +1,212 @@
+    {
+        throw  [System.IO.FileNotFoundException] "Could not find the SonarQube issue report at $sonarReportFilePath. Unable to post issues to the PR."
+    }
+
+    Write-Verbose "Report file found at $sonarReportFilePath"    
+        
+    CreateProjectGuidToProjectPathMap    
+    
+    $sonarReportFilePath = GetSonarReportFilePath
+    $json = DeserializeReport $sonarReportFilePath
+    CreateComponentKeyToPathMap $json
+
+    # '@' makes sure the result set is returned as an array
+    $newIssues = @($json.issues | Where {$_.isNew -eq $true})
+    Write-Host "Found $($json.issues.Count) issues out of which $($newIssues.Count) are new"
+
+    $newIssues = AnnotateIssuesWithRelativePath $newIssues
+    
+    return $newIssues
+}
+
+#endregion
+
+#region Private
+
+#
+# Returns the path to the file containing the issues that SonarQube generates when ran in issues / incremental mode
+#
+function GetSonarReportFilePath
+{ 
+    $sonarReportFolderPath = GetSonarScannerDirectory
+    $sonarReportFilePath = [System.IO.Path]::Combine($sonarReportFolderPath, "sonar-report.json")
+
+    return $sonarReportFilePath
+}
+
+function DeserializeReport
+{
+    param ([ValidateNotNullOrEmpty()][string]$sonarReportFilePath)
+    
+    Add-Type -AssemblyName "System.Web.Extensions"
+    
+    # Read sonar-report.json file as a json object
+    $sonarReportFileContent = Get-Content -Raw $sonarReportFilePath
+    $jsonSer = New-Object -TypeName System.Web.Script.Serialization.JavaScriptSerializer
+    
+    # Default value of MaxJsonLength is 2,097,152. Increasing max length to a 
+    $jsonSer.MaxJsonLength = [int]::MaxValue
+    $json = $jsonSer.DeserializeObject($sonarReportFileContent)   
+    
+    return $json 
+}
+
+#
+# Creates a mapping between key and path for componets (a SonarQube concept that translates to msbuild projects)
+#
+function CreateComponentKeyToPathMap($json)
+{
+    foreach ($component in $json.components)
+    {
+        if (!$ComponentKeyAndPathMap.ContainsKey($component.key))
+        {
+            $ComponentKeyAndPathMap.Add($component.key, $component.path)
+        }
+    }
+}
+
+#
+# Creates a mapping of msbuild project guid and the path of .xxproj file on disk using ProjectInfo.xml file
+#
+function CreateProjectGuidToProjectPathMap
+{        
+    $parentFolder = GetSonarQubeOutDirectory
+    $parentFolderItem = Get-Item $parentFolder
+    $directories = $parentFolderItem.GetDirectories()
+
+    foreach ($directory in $directories)
+    {
+        $projectInfoFilePath = [System.IO.Path]::Combine($directory.FullName, "ProjectInfo.xml")
+        
+        if ([System.IO.File]::Exists($projectInfoFilePath))
+        {
+            Write-Verbose "CreateProjectGuidToProjectPathMap: Processing project info file: $projectInfoFilePath"
+            [xml]$xmlContent = Get-Content $projectInfoFilePath
+
+            if ($xmlContent -and !$ProjectGuidAndFilePathMap.ContainsKey($xmlContent.ProjectInfo.ProjectGuid))
```

Comment:
Should we log a warning when ProjectGuidAndFilePathMap doesn't have xmlContent.ProjectInfo.ProjectGuid?